### PR TITLE
feat/init route

### PR DESCRIPTION
**Pull request by kyteinsky**

### CODE REVIEW

**Review by bigcat88 — COMMENTED** [review]
where is the call to "path=f'/index.php/apps/app_api/apps/status/{getenv("APP_ID")}'" after each model download?

File: context_chat_backend/controller.py [hunk 1]
```diff
@@ -96,7 +75,21 @@ def _():
 	return JSONResponse(content={'status': 'ok'}, status_code=200)
 
 
+@app.post('/init')
+async def _():
+	def update_progress(progress: int):
+		ocs_call(
+			method='PUT',
+			path=f'/index.php/apps/app_api/apps/status/{getenv("APP_ID")}',
+			json_data={ 'progress': max(100, progress) },
```
[the remark this answers is not in the document]
Comment (kyteinsky) [reply context_chat_backend/controller.py:84]: oops, meant to do `min`, sorry

**Review by andrey18106 — CHANGES_REQUESTED** [review]
Didn't succeed with local manual install test - looks like the ExApp webserver is hanged after it and `/init` doesn't happen to report to Nextcloud `progress=100`.

UPD: Maybe I missed something, let me retry to verify
UPD2: No, unfortunately not working, ocs init dispatch call should be async and not block the python process

**Comment by kyteinsky** [comment]
> Didn't succeed with local manual install test - looks like the ExApp webserver is hanged after it and `/init` doesn't happen to report to Nextcloud `progress=100`.

yeah noticed that the ocs call wasn't going through. I was looking through the nc_py_api to see how it handled this and seems like the issue is related to ocs call not being async here.

**Comment by kyteinsky** [comment]
made the call async and the url was wrong (maybe old path) :p

**Comment by kyteinsky** [comment]
while we are here, would you mind also testing the persistent storage @andrey18106 ? Thank you!

File: context_chat_backend/controller.py [hunk 2]
```diff
@@ -96,7 +75,26 @@ def _():
...
+async def _():
+	async def update_progress(progress: int):
+		await ocs_call(
+			method='PUT',
+			path=f'/ocs/v1.php/apps/app_api/apps/status/{getenv("APP_ID")}',
+			json_data={ 'progress': min(100, progress) },
+		)
+
+	if not app.extra.get('ENABLED', False):
+		await download_all_models(app, update_progress)
```
Comment (bigcat88) [inline context_chat_backend/controller.py:88]: please use FastAPI BackgroundTasks in which you will download models in the background

Python's `async` is a blocking call in this context, reply for "/init" will not be received on NC side..

Comment (andrey18106) [reply context_chat_backend/controller.py:88]: Yes, can confirm this issue, finally received time-out error on `/init`

Comment (kyteinsky) [reply context_chat_backend/controller.py]: that did the trick! How to know if async is blocking in a certain context though?

**Review by andrey18106 — APPROVED** [review]

**Pull request merged**